api/grpc/interceptor/AuthnInterceptor.go

```diff
@@ -63,6 +63,28 @@ func NewAuthnInterceptor(configs []api.AuthConfig) (*AuthnInterceptor, error) {
 	return &AuthnInterceptor{providers}, nil
 }
 
+func validateAuthConfigs(configs []api.AuthConfig) (err error) {
+	if configs == nil {
+		return fmt.Errorf("configs should not be nil")
+	}
+
+	checked := make(map[api.AuthConfig]bool)
+
+	for _, config := range configs {
+		if config == (api.AuthConfig{}) {
+			return fmt.Errorf("authnconfig should not be zero value")
```

Comment:
should not be "nil" ? Also, the check above can be "nil". empty objects checks are same as nil checks